### PR TITLE
CI improvements

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
Changes:
- CI file is now called `qa.yaml`, bringing it in line with other Ubuntu and Canonical projects
- Tests now have the `shuffle` flag enabled.
- Splits the CI workflow in two: `qa.yaml` and `qa-azure.yaml`. This way the mock tests can be outside the concurrency group.
- Cleans up the steps a bit to make them consistent between QA and QA-Azure
- Enables the race flag! This is a bugfix as we hitherto ran the "race" tests without the flag enabled 🤦

### CODE REVIEW

File: .github/workflows/qa.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    - name: Lint with real back-end
+      uses: golangci/golangci-lint-action@v3
+      with:
+        version: v1.52.0
+    - name: Prepare repo
+      shell: bash
+      run: |
+        # Prepare repo with mock rootfs and run mock tests
+        mkdir -p images
+        touch images/empty.tar.gz
```
Comment (didrocks) [inline .github/workflows/qa.yaml:50]: I don’t remember why we need those for the mock? It seems that we shouldn’t to me, as we are not going to use the real WSL layer. At worse, the mock can create them when being instantiated and pointing tests to it, wdyt?

**Review by didrocks — COMMENTED** [review]
One question but overall, looks really good :)

File: .github/workflows/qa.yaml (hunk 1 above)
Comment (EduardGomezEscandell) [reply .github/workflows/qa.yaml:50]: If the rootfs does not exist, the wslApi syscall fails without explaining why, so I added a check before calling the back-end to make the error less obscure (similar to checking if the distro exists before registering). The point is, this check is in the front-end of GoWSL.

I could either move it to the back-end, or have the mock tests create the file. I think the first option makes more sense, but I'd do it in a diferent PR as this goes beyond enhancing the CI.

Comment (didrocks) [reply .github/workflows/qa.yaml:50]: Do you mind opening a bug report for this? I agree, let’s proceed with this one.

**Review by didrocks — APPROVED** [review]

File: .github/workflows/qa.yaml (hunk 1 above)
Comment (EduardGomezEscandell) [reply .github/workflows/qa.yaml:50]: Nevermind, I think putting it in TestMain is the best option, this way `TestRegister` requires no modification. I'll do it in a follow-up PR.

**Pull request merged**